### PR TITLE
Analytics Configuration - Code Refactoring

### PR DESCRIPTION
**Pull request by rmsamitha**

### CODE REVIEW

File: modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java [hunk 1]
```diff
@@ -47,35 +47,31 @@
...
+        }
+        String bpsUrl = DASConfigurationUtils.getBPSURL();
         RegistryUtils.setTrustStoreSystemProperties();
         HttpClient httpClient = new HttpClient();
-        String requestUrl = bpsurl + AnalyticsConfigConstants.BPS_PROCESS_VAR_PUBLISH_REST_PATH + processName + "_"
+        String requestUrl = bpsUrl + AnalyticsConfigConstants.BPS_PROCESS_VAR_PUBLISH_REST_PATH + processName + "_"
                 + processVersion;
 
         PostMethod postRequest = new PostMethod(requestUrl);
         postRequest.setRequestHeader("Authorization", DASConfigurationUtils.getAuthorizationHeader());
```
Comment (vinok88) [inline modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java]: Should use a constant here.

File: modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java [hunk 2]
```diff
@@ -85,19 +81,17 @@ public static void post(String dasConfigDetails, String processName, String proc
...
             }
         }
-
         //deal with the response
         if (returnCode == HttpStatus.SC_OK) {
-            log.info("BPS was acknowleged the Analytics Configuration details");
-        } else if (returnCode == HttpStatus.SC_INTERNAL_SERVER_ERROR) {
-            log.info(responseMsg);
-            throw new RuntimeException(responseMsg);
+            if (log.isDebugEnabled()) {
```
Comment (vinok88) [inline modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java]: If check is not required if there is no string manipulation in the message.

File: modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java [hunk 3]
```diff
@@ -85,19 +81,17 @@ public static void post(String dasConfigDetails, String processName, String proc
...
+                log.debug("BPS was configured with analytics configuration details.");
+            }
         } else {
             String errMsg =
                     "Failed : Sending the REST Post call to the WSO2 BPS to communicate the analytics configuration "
                             + "details to the BPS from PC\n: HTTP Error code : "
                             + returnCode;
-            log.info(errMsg);
+            log.error(errMsg);
```
Comment (vinok88) [inline modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java]: No need to log the error here.

File: modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java (hunk 2 above)
Comment (rmsamitha) [inline modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java]: Fixed with #164

File: modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java (hunk 3 above)
Comment (rmsamitha) [inline modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java]: Fixed with #164

File: modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java [hunk 4]
```diff
@@ -47,35 +47,31 @@
     public static void post(String dasConfigDetails, String processName, String processVersion)
             throws IOException, XMLStreamException, RuntimeException {
 
-        String bpsurl = DASConfigurationUtils.getBPSURL();
+        if(log.isDebugEnabled()){
```
Comment (darshanasbg) [inline modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/BPSConfigRestClient.java]: Fix formatting issues.

File: modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/utils/DASConfigurationUtils.java [hunk 5]
```diff
@@ -129,19 +131,19 @@ public static void setPropertyDASAnalyticsConfigured(String processName, String
...
+                        log.debug("isDasConfigedForAnalytics property in process.rxt is set as true");
                     }
                 }
             }
         } catch (TransformerException | RegistryException e) {
-            String errMsg = "Exception in setting property isDASAnalyticsConfigured in process.rxt for the process:"
+            String errMsg = "Error in setting property isDASAnalyticsConfigured in process.rxt for the process: "
                     + processName + ":" + processVersion;
             log.error(errMsg, e);
             throw new ProcessCenterException(errMsg, e);
```
Comment (darshanasbg) [inline modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/utils/DASConfigurationUtils.java]: Why both log and throw? Shouldn't it have only one?

Comment (rmsamitha) [inline modules/components/analytics/core/kpi/org.wso2.carbon.pc.analytics.core.kpi/src/main/java/org/wso2/carbon/pc/analytics/core/kpi/utils/DASConfigurationUtils.java]: Thanks @darshanasbg This is already fixed with 636d316f1c9c92034a9d3c7c1a8fef26dbd5ba1b

**Pull request merged**